backend/workflow_manager/workflow_v2/execution.py

```diff
@@ -171,6 +175,26 @@ def get_execution_instance(self) -> WorkflowExecution:
         )
         return execution
 
+    @classmethod
+    def get_execution_instance_by_id(
+        cls, execution_id: str
+    ) -> Optional[WorkflowExecution]:
+        """Get execution by execution ID.
+
+        Args:
+            execution_id (str): UID of execution entity
+
+        Returns:
+            Optional[WorkflowExecution]: WorkflowExecution Entity
+        """
+        try:
+            execution: WorkflowExecution = WorkflowExecution.objects.get(
+                pk=execution_id
+            )
+            return execution
+        except WorkflowExecution.DoesNotExist:
+            return None
```

Comment:
NIT: Might have missed this comment on the v1 PR but do you think it'll be good to log this case @vishnuszipstack @muhammad-ali-e ?

Reply:
@chandrasekharan-zipstack No, It is just a validation, That the execution already exist or not. We can make the logic some what better if possible.